js/src/components/tree-select-control/list.js

```diff
@@ -1,250 +1,87 @@
-				}
-				setExpanded( false );
-				break;
-
-			default:
-		}
-	}
-
-	toggleKeyEvents( isListening ) {
-		const { node } = this.props;
-		// This exists because we must capture ENTER key presses before RichText.
-		// It seems that react fires the simulated capturing events after the
-		// native browser event has already bubbled so we can't stopPropagation
-		// and avoid RichText getting the event from TinyMCE, hence we must
-		// register a native event handler.
-		const handler = isListening
-			? 'addEventListener'
-			: 'removeEventListener';
-		node[ handler ]( 'keydown', this.handleKeyDown, true );
-	}
+const List = ( { options = [], value = [], onChange = () => {} } ) => {
+	return (
+		<div
+			className="woocommerce-tree-select-control__listbox"
+			role="listbox"
+			tabIndex="-1"
+		>
+			<Options
+				options={ options }
+				value={ value }
+				onChange={ onChange }
+			/>
+		</div>
+	);
+};
 
-	componentDidMount() {
-		const { selectedIndex } = this.props;
-		if ( selectedIndex > -1 ) {
-			this.scrollToOption( selectedIndex );
+const Options = ( { options = [], value, parent = '', onChange } ) => {
+	/**
+	 * Returns true if all the children for the parent are selected
+	 *
+	 * @param {Option} option The parent option to check
+	 */
+	const isParentSelected = ( option ) => {
+		if ( ! option.children ) {
+			return false;
 		}
-		this.toggleKeyEvents( true );
-	}
-
-	componentWillUnmount() {
-		this.toggleKeyEvents( false );
-	}
 
-	render() {
-		const {
-			instanceId,
-			listboxId,
-			options,
-			selectedIndex,
-			staticList,
-		} = this.props;
-		const listboxClasses = classnames(
-			'woocommerce-select-control__listbox',
-			{
-				'is-static': staticList,
-			}
+		return option.children.every(
+			( child ) => value.includes( child.id ) || isParentSelected( child )
 		);
+	};
+
+	const hasChildren = ( option ) => !! option.children?.length;
 
+	return options.map( ( option ) => {
 		return (
 			<div
-				ref={ this.listbox }
-				id={ listboxId }
-				role="listbox"
-				className={ listboxClasses }
-				tabIndex="-1"
+				key={ `${ parent }-${ option.id }` }
+				className="woocommerce-tree-select-control__group"
 			>
-				{ options.map( ( option, index ) => (
-					<Button
-						ref={ this.getOptionRef( index ) }
-						key={ option.key }
-						id={ `woocommerce-select-control__option-${ instanceId }-${ option.key }` }
-						role="option"
-						aria-selected={ index === selectedIndex }
-						disabled={ option.isDisabled }
-						className={ classnames(
-							'woocommerce-select-control__option',
-							{
-								'is-selected': index === selectedIndex,
-							}
-						) }
-						onClick={ () => this.select( option ) }
-						tabIndex="-1"
-					>
-						{ option.label }
-					</Button>
-				) ) }
+				<CheckboxControl
+					value={ option.id }
+					className={ classnames(
+						'woocommerce-tree-select-control__option',
+						'woocommerce-tree-select-control__parent'
```

Comment:
For now, I removed the parent class since for now is useless. I will add the proper logic of this(used in the expander) in Milestone 2 https://github.com/woocommerce/google-listings-and-ads/pull/1243/commits/c40c5db47f603b0d25eaf64e3a31d29405a6ba04

Reply:
Sounds like a good plan. 🙂 👍